cli/internal/cmd/upgradeapply.go

```diff
@@ -258,15 +258,28 @@ func (u *upgradeApplyCmd) parseUpgradeVars(cmd *cobra.Command, conf *config.Conf
 		targets := []string{}
 
 		vars := &terraform.GCPClusterVariables{
-			CommonVariables: commonVariables,
-			Project:         conf.Provider.GCP.Project,
-			Region:          conf.Provider.GCP.Region,
-			Zone:            conf.Provider.GCP.Zone,
-			CredentialsFile: conf.Provider.GCP.ServiceAccountKeyPath,
-			InstanceType:    conf.Provider.GCP.InstanceType,
-			StateDiskType:   conf.Provider.GCP.StateDiskType,
-			ImageID:         imageRef,
-			Debug:           conf.IsDebugCluster(),
+			Name: conf.Name,
+			NodeGroups: map[string]terraform.GCPNodeGroup{
+				"control_plane_default": {
+					Role:            "ControlPlane",
+					StateDiskSizeGB: conf.StateDiskSizeGB,
+					Zone:            conf.Provider.GCP.Zone,
+					InstanceType:    conf.Provider.GCP.InstanceType,
+					DiskType:        conf.Provider.GCP.StateDiskType,
+				},
+				"worker_default": {
+					Role:            "Worker",
+					StateDiskSizeGB: conf.StateDiskSizeGB,
+					Zone:            conf.Provider.GCP.Zone,
+					InstanceType:    conf.Provider.GCP.InstanceType,
+					DiskType:        conf.Provider.GCP.StateDiskType,
+				},
+			},
+			Project: conf.Provider.GCP.Project,
+			Region:  conf.Provider.GCP.Region,
+			Zone:    conf.Provider.GCP.Zone,
+			ImageID: imageRef,
+			Debug:   conf.IsDebugCluster(),
 		}
```

Comment:
DRY and also use in create.go:

```suggestion
	pkg terraform
	func NewGCPClusterVariables(conf config.Config) *terraform.GCPClusterVariables {
		return &terraform.GCPClusterVariables{
			Name: conf.Name,
			NodeGroups: map[string]terraform.GCPNodeGroup{
				"control_plane_default": {
					Role:            "ControlPlane",
					StateDiskSizeGB: conf.StateDiskSizeGB,
					Zone:            conf.Provider.GCP.Zone,
					InstanceType:    conf.Provider.GCP.InstanceType,
					DiskType:        conf.Provider.GCP.StateDiskType,
				},
				"worker_default": {
					Role:            "Worker",
					StateDiskSizeGB: conf.StateDiskSizeGB,
					Zone:            conf.Provider.GCP.Zone,
					InstanceType:    conf.Provider.GCP.InstanceType,
					DiskType:        conf.Provider.GCP.StateDiskType,
				},
			},
			Project: conf.Provider.GCP.Project,
			Region:  conf.Provider.GCP.Region,
			Zone:    conf.Provider.GCP.Zone,
			ImageID: imageRef,
			Debug:   conf.IsDebugCluster(),
		}
	}
```

Reply:
sounds good

Reply:
Also as a general rule, I think we should follow 

> [A little copying is better than a little dependency.](https://www.youtube.com/watch?v=PAAkCSZUG1c&t=9m28s)

-- https://go-proverbs.github.io/

Reply:
Actually, since `upgradeapply.go` lives in `cli/internal/cmd` and `create.go` lives in  `cli/internal/cloudcmd`, I don't see an easy way to share this code.